quint/src/types/aliasInliner.ts

```diff
@@ -0,0 +1,87 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright (c) Informal Systems 2023. All rights reserved.
+ * Licensed under the Apache 2.0.
+ * See License.txt in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Inlining for type aliases, to be used when resolving type aliases with the
+ * lookup table is not feasible.
+ *
+ * @author Gabriela Moreira
+ *
+ * @module
+ */
+
+import { IRTransformer, transformDefinition, transformModule, transformType } from '../IRTransformer'
+import { LookupTable } from '../names/lookupTable'
+import { QuintDef, QuintModule } from '../quintIr'
+import { QuintType } from '../quintTypes'
+
+/**
+ * Replaces all type aliases in a QuintModule with their resolved types, using
+ * the provided LookupTable. Uninterpreted types are left unchanged.
+ *
+ * @param lookupTable - The LookupTable containing the type aliases to be
+ * resolved.
+ * @param quintModule - The QuintModule to transform.
+ *
+ * @returns The transformed QuintModule with all type aliases replaced with
+ * their resolved types.
+ */
+export function inlineAliases(lookupTable: LookupTable, quintModule: QuintModule): QuintModule {
```

Comment:
Purely stylistic, but I would flip the order of arguments here, given that the "main" argument is the module and the lookup table is mostly a helper.

```suggestion
export function inlineAliases(quintModule: QuintModule, lookupTable: LookupTable): QuintModule {
```